e2e-tests/single-machine/single_machine_deployment_validation.sh

```diff
@@ -0,0 +1,314 @@
+  TOTAL_TEST_PATIENTS=$(jq '.total' "${HOME_PATH}/${PARQUET_SUBDIR}/patients.json")
+  print_message "Total FHIR source test patients ---> ${TOTAL_TEST_PATIENTS}"
+
+  curl -L -X GET -u $fhir_username:$fhir_password --connect-timeout 5 --max-time 20 \
+    "${SOURCE_FHIR_SERVER_URL}${fhir_url_extension}/Encounter${query_param}" \
+    2>/dev/null >>"${HOME_PATH}/${PARQUET_SUBDIR}/encounters.json"
+  TOTAL_TEST_ENCOUNTERS=$(jq '.total' "${HOME_PATH}/${PARQUET_SUBDIR}/encounters.json")
+  print_message "Total FHIR source test encounters ---> ${TOTAL_TEST_ENCOUNTERS}"
+
+  curl -L -X GET -u $fhir_username:$fhir_password --connect-timeout 5 --max-time 20 \
+    "${SOURCE_FHIR_SERVER_URL}${fhir_url_extension}/Observation${query_param}" \
+    2>/dev/null >>"${HOME_PATH}/${PARQUET_SUBDIR}/obs.json"
+  TOTAL_TEST_OBS=$(jq '.total' "${HOME_PATH}/${PARQUET_SUBDIR}/obs.json")
+  print_message "Total FHIR source test obs ---> ${TOTAL_TEST_OBS}"
+}
+
+#######################################################################
+# Function to send command to pipeline controller to start pipeline.
+# Globals:
+#   PIPELINE_CONTROLLER_URL
+# Arguments:
+#   isFullRun: flag to indicate whether to start full or incremental run.
+#######################################################################
+function run_pipeline() {
+  isFullRun=$1
+  curl --location --request POST "${PIPELINE_CONTROLLER_URL}/run?isFullRun=${isFullRun}" \
+  --connect-timeout 5 \
+  --header 'Content-Type: application/json' \
+  --header 'Accept: */*'
+}
+
+#######################################################################
+# Function to check periodically parquet files in the given directory
+# and verify the number of created resources against the number got from
+# hapi server.
+# Globals:
+#   HOME_PATH
+#   PARQUET_SUBDIR
+#   TOTAL_TEST_PATIENTS
+#   TOTAL_TEST_ENCOUNTERS
+#   TOTAL_TEST_OBS
+# Arguments:
+#   isIncremental: flag to differentiate behavior between incremental and batch runs.
+#######################################################################
+function check_parquet() {
+  isIncremental=$1
+  runtime="5 minute"
+  end_time=$(date -ud "$runtime" +%s)
+  output="${HOME_PATH}/${PARQUET_SUBDIR}"
+  timeout=true
+
+  if [[ "${isIncremental}" == "true" ]]
+  then
+    # In case of incremental run, we will have two directories
+    # assuming batch run was executed before this.
+    TOTAL_TEST_PATIENTS=$((2*TOTAL_TEST_PATIENTS + 1))
+    TOTAL_TEST_ENCOUNTERS=$((2*TOTAL_TEST_ENCOUNTERS))
+    TOTAL_TEST_OBS=$((2*TOTAL_TEST_OBS))
+    print_message "Full batch run started."
+  else
+    print_message "Incremental run started."
+  fi
+
+  while [[ $(date -u +%s) -le $end_time ]]
+  do
+    # check whether output directory has started receiving parquet files.
+    if [ "$(ls -A $output)" ]
+    then
+      total_patients=$(java -jar ./parquet-tools-1.11.1.jar rowcount "${output}/*/Patient/" | awk '{print $3}')
```

Comment:
We limit to 120 chars I think.

Reply:
Hmm, do we? I mean do we have any style-checker running on our `.sh` files? I think the problem is that because we don't have the automatic check, we have not been consistent.

Reply:
Yeah for Java it is 120 but for bash it is 80 (in both internal and external style guides AFAIK); I would not bee too strict but maybe just a few more characters over 80 if it saves us a line.

Reply:
Oh ok. I will try to incorporate bash style checker as part of spotless soon. Resolving this for now.